braintree.go

```diff
@@ -116,6 +116,11 @@ func (g *Braintree) execute(method, path string, xmlObj interface{}) (*Response,
 		g.Logger.Printf("<\n%s", string(btr.Body))
 	}
 
+	err = btr.httpError()
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
A side effect of putting `httpError` ahead of `apiError` is that if braintree includes error details for any of the status codes it won't get serialized into a `BraintreeError` any longer. Looking at the [Braintree Java SDK](https://github.com/braintree/braintree_java/blob/be892f293346a7d9a77bda750976cebdb1ce0c6b/src/main/java/com/braintreegateway/util/Http.java#L275-L300) there is at least one other case it seems where a message will be included. So it's probably safer to attempt to parse the response for a `BraintreeError` and only after that fails, if the status code is `> 299`, return your new `httpError` instead of the `fmt.Errorf`.